zwrap/sample.go

```diff
@@ -98,70 +103,62 @@ func (s *sampler) With(fields ...zap.Field) zap.Logger {
 }
 
 func (s *sampler) Check(lvl zap.Level, msg string) *zap.CheckedMessage {
-	if !s.check(lvl, msg) {
-		return nil
+	cm := s.Logger.Check(lvl, msg)
+	switch lvl {
+	case zap.PanicLevel, zap.FatalLevel:
+		return cm
 	}
-	return zap.NewCheckedMessage(s.Logger, lvl, msg)
+	if s.sampled(msg) {
+		return cm
+	}
+	return nil
 }
 
 func (s *sampler) Log(lvl zap.Level, msg string, fields ...zap.Field) {
-	if s.check(lvl, msg) {
+	if s.Logger.Check(lvl, msg) != nil && s.sampled(msg) {
```

Comment:
Hmm, it certainly will be the simplest change (and in line with the current doc strings) for now to make `Log`ing at fatal or panic level not sampled.

However, now that we're in this place of "you _can_ log at Fatal or Panic level directly, and it's not terminal", a sampled logger should now downsample such logging to protect downstreams.  That said, I can see the argument for "no, we want that to blow up as horrible unsupported/broken usage"...